site/src/pages/WorkspacesPage/WorkspacesPage.tsx

```diff
@@ -129,3 +168,106 @@ const useWorkspacesFilter = ({
     },
   }
 }
+
+const BatchDeleteConfirmation = ({
+  checkedWorkspaces,
+  open,
+  onClose,
+  onDelete,
+}: {
+  checkedWorkspaces: Workspace[]
+  open: boolean
+  onClose: () => void
+  onDelete: () => void
+}) => {
+  const [confirmValue, setConfirmValue] = useState("")
+  const [confirmError, setConfirmError] = useState(false)
+  const [isDeleting, setIsDeleting] = useState(false)
+
+  const close = () => {
+    if (isDeleting) {
+      return
+    }
+
+    onClose()
+    setConfirmValue("")
+    setConfirmError(false)
+    setIsDeleting(false)
+  }
+
+  const confirmDeletion = async () => {
+    setConfirmError(false)
+
+    if (confirmValue.toLowerCase() !== "delete") {
```

Comment:
We could consider only accepting the value as stated (i.e. must be upper-case `DELETE`) to confirm. But there may be differing opinions on this.